Marketplace/AzureStack.Marketplace.psm1

```diff
@@ -0,0 +1,194 @@
+        [String] $galleryItemBlobURI,
+
+        [Parameter(Mandatory=$true, ParameterSetName='GalleryItemFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='GalleryItemFromAzure')]
+        [ValidateNotNullorEmpty()]
+        [String] $tenantID,
+
+        [Parameter(ParameterSetName='GalleryItemFromLocal')]
+        [Parameter(ParameterSetName='GalleryItemFromAzure')]
+        [String] $location = 'local',
+
+        [Parameter(Mandatory=$true, ParameterSetName='GalleryItemFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='GalleryItemFromAzure')]
+        [System.Management.Automation.PSCredential] $azureStackCredentials,
+
+        [Parameter(ParameterSetName='GalleryItemFromLocal')]
+        [Parameter(ParameterSetName='GalleryItemFromAzure')]
+        [string] $ArmEndpoint = 'https://adminmanagement.local.azurestack.external'
+    )
+
+    $resourceGroupName = "addiresourcegroup"
+    $storageAccountName = "addgistorageaccount"
+    $containerName = "addgicontainer"
+
+    $subscription = (Get-AzureStackAdminSubscription -TenantId $tenantId -AzureStackCredentials $azureStackCredentials -ArmEndpoint $ArmEndpoint)
+
+    #potentially the RG was not cleaned up when exception happened in previous run. Test for exist
+    if (-not (Get-AzureRmResourceGroup -Name $resourceGroupName -Location $location -ErrorAction SilentlyContinue)) {
+        New-AzureRmResourceGroup -Name $resourceGroupName -Location $location 
+    }
+
+    #same for storage
+    $storageAccount = Get-AzureRmStorageAccount -Name $storageAccountName -ResourceGroupName $resourceGroupName -ErrorAction SilentlyContinue
+    if (-not ($storageAccount)) {
+        $storageAccount = New-AzureRmStorageAccount -Name $storageAccountName -Location $location -ResourceGroupName $resourceGroupName -Type Standard_LRS
+    }
+
+    Set-AzureRmCurrentStorageAccount -StorageAccountName $storageAccountName -ResourceGroupName $resourceGroupName
+    
+    #same for container
+    $container = Get-AzureStorageContainer -Name $containerName -ErrorAction SilentlyContinue
+    if (-not ($container)) {
+        $container = New-AzureStorageContainer -Name $containerName -Permission Blob
+    }
+
+    if($pscmdlet.ParameterSetName -eq "GalleryItemFromLocal") {
+        $storageAccount.PrimaryEndpoints.Blob
+        $script:galleryItemName = Split-Path $galleryItemLocalPath -Leaf
+        $script:galleryItemBlobURIFromLocal = '{0}{1}/{2}' -f $storageAccount.PrimaryEndpoints.Blob.AbsoluteUri, $containerName,$galleryItemName
+        Set-AzureStorageBlobContent -File $galleryItemLocalPath -Container $containerName -Blob $galleryItemName
+        $galleryItemBlobURI = $galleryItemBlobURIFromLocal
+    }
+
+    if((Get-Module AzureStack).Version -ge [System.Version] "1.2.9"){
+        Add-AzureRMGalleryItem -GalleryItemUri $galleryItemBlobURI
+    }else{
+        Add-AzureRMGalleryItem -SubscriptionId $subscription -GalleryItemUri $galleryItemBlobURI -ApiVersion 2015-04-01
```

Comment:
Just want to confirm, is this by design that the previous call of Add-AzureRMGalleryItem in line 78 doesn't needs $subscription?